### PR TITLE
Bug 1574651 - a rootUrl per repo

### PR DESCRIPTION
**Pull request by djmitche**
I'll do this bug in phases.  The first phase is here: set a rootUrl for each repo, and use that to generate URLs, etc. in the frontend.  For the moment this is mostly a no-op, since the user is only logged into one rootUrl, but that will change soon.

Subsequent phases:
 * ingest from multiple pulse exchanges, each for a different rootUrl
 * (not this bug) implement third-party sign-in, in such a way that Treeherder can hold user creds for multiple TC deployments at the same time

### CODE REVIEW

**Comment by djmitche** [comment]
I tested the frontend changes manually, and for the threading of currentRepo through all those components I used the React devtools to see that the prop was set everywhere.

**Comment by camd** [comment]
I ran out of time today.  I'll review on Tuesday when we're all back.  Thanks Dustin!

File: ui/models/job.js [hunk 1]
```diff
@@ -83,7 +83,7 @@ export default class JobModel {
     return JobModel.getList(options, config);
   }
 
-  static async retrigger(jobs, repoName, notify, times = 1) {
+  static async retrigger(jobs, repoName, notify, currentRepo, times = 1) {
```
Comment (camd) [inline ui/models/job.js]: Clearly I have some code cleanup to do here...  :)  ``repoName`` is never used.  But you're right to just leave it here.  I'll take care of it in a follow-up PR.

**Comment by camd** [comment]
Thanks for doing this @djmitche !  :)  I've reviewed the code and it looks great.  But, my humble apologies for the state of the ``repoName`` param all over the place.  I was in the middle of refactoring much of it's use out when I got pulled away onto a couple other things.  Sorry you had to deal with that!

I want to test this a bit before I merge it.  I will push it up to our prototype server tomorrow and ask the sheriffs to test some of the workflows as well.

Thanks again!

File: ui/models/job.js (hunk 1 above)
Comment (djmitche) [reply ui/models/job.js]: Don't apologize, it's not bad :)  The cleanup was because I had *added* a new parameter that made the old one redundant.

**Comment by djmitche** [comment]
@camd rebased this in #5405.  The next phase is in #5399.

**Pull request closed without merging**